src/Services/PortfolioPlanningDataService.ts

```diff
@@ -560,23 +656,52 @@ export class ODataQueryBuilder {
      * @param input 
      */
     private static BuildODataDescendantsQuery(input: PortfolioPlanningQueryInput): string {
-        const requirementWiTypes = input.RequirementWorkItemTypes.map(id => `WorkItemType eq '${id}'`);
+        const descendantsTypesSet: { [workItemType: string]: string } = {};
+        const requirementWiTypes: string[] = [];
+
+        input.WorkItems.forEach(wi => {
+            const wiTypeKey = wi.DescendantsWorkItemTypeFilter.toLowerCase();
+
+            if (!descendantsTypesSet[wiTypeKey]) {
+                descendantsTypesSet[wiTypeKey] = wi.EffortODataColumnName;
+                requirementWiTypes.push(`WorkItemType eq '${wiTypeKey}'`);
+            }
+        });
+
+        const effortSelectionConditional: string = this.BuildEffortSelectionConditional(descendantsTypesSet);
 
+        // prettier-ignore
         return (
             "Descendants(" +
-            "$apply=" +
-            `filter(${requirementWiTypes.join(" or ")})` +
-            "/aggregate(" +
-            "$count as TotalCount," +
-            "iif(StateCategory eq 'Completed',1,0) with sum as CompletedCount," +
-            "StoryPoints with sum as TotalStoryPoints," +
-            "iif(StateCategory eq 'Completed',StoryPoints,0) with sum as CompletedStoryPoints" +
-            ")" +
-            "/compute(" +
-            "(CompletedCount div cast(TotalCount, Edm.Decimal)) as CountProgress," +
-            "(CompletedStoryPoints div TotalStoryPoints) as StoryPointsProgress" +
-            ")" +
+                "$apply=" +
+                    `filter(${requirementWiTypes.join(" or ")})` +
+                    "/aggregate(" +
+                        "$count as TotalCount," +
+                        "iif(StateCategory eq 'Completed',1,0) with sum as CompletedCount," +
+                        `${effortSelectionConditional} with sum as TotalEffort,` +
+                        `iif(StateCategory eq 'Completed',${effortSelectionConditional},0) with sum as CompletedEffort` +
+                    ")" +
+                    "/compute(" +
+                        "(CompletedCount div cast(TotalCount, Edm.Decimal)) as CountProgress," +
+                        "(iif(TotalEffort gt 0, CompletedEffort div TotalEffort, 0)) as EffortProgress" +
+                    ")" +
             ")"
         );
     }
+
+    private static BuildEffortSelectionConditional(input: { [workItemType: string]: string }): string {
+        const keys = Object.keys(input);
+        if (keys.length === 0) {
+            return "0";
+        }
+
+        const workItemType = keys[0];
+        const oDataColumnName = input[workItemType];
+
+        delete input[workItemType];
+
+        return `iif(WorkItemType eq '${workItemType}', ${oDataColumnName}, ${this.BuildEffortSelectionConditional(
```

Comment:
Odata is not supporting nested `IIF`s in the `aggregate`... need to change the approach... will send update tomorrow.